### PR TITLE
Cirrus: Remove hardcoded snap version

### LINKED ISSUE #4134: Cirrus-CI: test_building_snap and upload_snap produce podman 0.11.1.1
/kind bug

**Description**

In `.cirrus.yml` these two tasks build, then build + upload podman to snapcraft.  For some reason both believe the podman version is `0.11.1.1`

**Steps to reproduce the issue:**

1. Open PR

2. Merge PR

**Describe the results you received:**

(from cirrus output)
```
...cut...
'grade' property not specified: defaulting to 'stable'
Snapping 'podman' ...
Snapped podman_0.11.1.1_amd64.snap
Downloading 'v0.11.1.1.tar.gz'                                                 
Downloading 'v0.11.1.1.tar.gz'                                                 
Preparing to push 'podman_0.11.1.1_amd64.snap'.
After pushing, an attempt will be made to release to 'edge'
...cut...
```

**Describe the results you expected:**

The proper version should be built and uploaded

### PR DESCRIPTION
**Pull request by abitrolly**
Should fix #4134.

`git` version semantics is described at https://snapcraft.io/docs/snapcraft-yaml-reference

### CODE REVIEW

**Comment by abitrolly** [comment]
I run out of ideas. Asked here https://forum.snapcraft.io/t/the-go-plugin/8505/7?u=abitrolly

**Comment by cevich** [comment]
(re-ran failed test snap task)

**Comment by abitrolly** [comment]
@cevich it won't help. It is either `snapcraft` go plugin not setting all paths correctly, or it could be that `libpod` biuld is more complicated and needs more processing than required by standard `go` builds supported by `snapcraft`.

The build from tagged source was successful, and I think the problem is with `snapcraft` and source path. The workaround is to reuse the binary from the build task, but I don't see how to do this yet in Cirrus CI.

**Comment by cevich** [comment]
@abitrolly thanks for taking a look.  Hmmm, so `snapcraft` must not be using the Makefile then?

Re: packaging based on the release zip/tar files we upload for each testing task.  It is possible to retrieve them during either/both `test_building_snap` and `upload_snap` (assuming some `depends_on` adjustments so they happen right before 'success').

For both for PRs and on 'master', we use fixed base URLs (see `<BRANCH  |  PR>` below)

* `https://storage.googleapis.com/libpod-pr-releases/` for PRs
* `https://storage.googleapis.com/libpod-master-releases/` for master

The filenames uploaded for any build can be discovered programmaticly, given a clean tree (no added/removed commits).  The file names use the format:

```
<BASENAME | REMOTENAME>-<VERSION>-<BRANCH  |  PR>-<DIST>-<ARCH>.gz
```

Where:
* `<BASENAME | REMOTENAME>` is the value from `hack/get_release_info.sh` with one of those keywords as an argument.  Currently "podman" or "podman-remote"
* `<VERSION>` is the value from `hack/get_release_info.sh VERSION`
* `<BRANCH>` is the value of `$CIRRUS_BRANCH` (if `$CIRRUS_PR` is an empty string)
* `<PR>` is the string 'pr' followed by the value of '$CIRRUS_PR` (if NOT an empty string)
* `<DIST>` is the value from `hack/get_release_info.sh DIST`.  Currently "fedora", "ubuntu", "windows" or "darwin"
* `<ARCH>` is the value from `hack/get_release_info.sh ARCH`

Alternatively, all the above (except `<BRANCH | PR>`) can be parsed from the `make release.txt` output file.

HTH

**Comment by vrothberg** [comment]
@cevich @abitrolly are you still working on this PR?

**Comment by cevich** [comment]
@vrothberg I think neither of us has an idea how to fix it?  Maybe we should disable snap packaging?

**Comment by abitrolly** [comment]
@cevich the right fix is to grab the binaries and pack them into `snap` instead of rebuilding the project from scratch using `snapcraft`. If you pass and echo these binary URLs in `upload_snap` task, I can see what it takes to pass them to `snapcraft`.

**Comment by cevich** [comment]
@abitrolly thanks for the hint, I simply have to be cautious about where I spent my time.  Currently my automation-related priorities are keeping me busy for the foreseeable feature.  Though I would like to see this effort move forward (in general), and I'm happy to consult as required.

@lsm5 even if you can't, do you know anyone that can help with this?

**Comment by rhatdan** [comment]
Perhaps @abitrolly Would like to take a stab.  Hint Hint.

**Comment by abitrolly** [comment]
@cevich `make` build fails. What does it need? https://github.com/containers/libpod/pull/4135/checks?check_run_id=359532249

**Comment by cevich** [comment]
@abitrolly sorry I missed your mention above.   TBH I'm not sure at all what that means :confused: but it would be worth rebasing this PR against the latest master to see if it reproduces.

**Comment by abitrolly** [comment]
@cevich snap uses `go 1.10.4` - could it be the problem?

https://cirrus-ci.com/task/6604298431758336?command=snapcraft#L583

**Comment by cevich** [comment]
> snap uses go 1.10.4 - could it be the problem?

Easily...along with a whole host of other build-environment issues :disappointed:   Not sure how much trouble it will be (docs may not be accurate): https://github.com/containers/libpod/blob/master/install.md#building-from-scratch

The setup used for preparing ubuntu VM is in: `./contrib/cirrus/packer/ubuntu_setup.sh`
(minus installing the podman package)  There's also a `Dockerfile` which may be more up to date WRT environment setup.

Could either of those be re-used, maybe even with some modifications to work for both usages?

**Comment by abitrolly** [comment]
`gate` linter fails on go code and I don't touch it anymore.

File: contrib/snapcraft/snap/snapcraft.yaml [hunk 1]
```diff
@@ -13,14 +13,15 @@ base: core18
...
+    plugin: make
+    source: ../..
+    build-snaps:
+     - go/1.13/stable
     build-packages:
      # https://github.com/containers/libpod/blob/master/install.md#build-and-run-dependencies
      - btrfs-tools
      - git
-     - golang-go
+     #- golang-go
```
Comment (giuseppe) [inline contrib/snapcraft/snap/snapcraft.yaml]: if it is not needed please drop it instead of commenting it out

Comment (abitrolly) [reply contrib/snapcraft/snap/snapcraft.yaml]: Dropped. It doesn't help failing `gating` still.

**Comment by cevich** [comment]
There was a problem introduced last week that affected the gate job for everybody.  A fix was pushed to master, so perhaps you just need to rebase this PR?

**Comment by cevich** [comment]
This probably needs a rebase too

**Comment by vrothberg** [comment]
Friendly ping.

**Comment by cevich** [comment]
@abitrolly suggestion/idea/question: [Maybe this could help some duplication](https://github.com/containers/libpod/blob/master/contrib/cirrus/packer/ubuntu_packaging.sh) I recent merge started using this in our Dockerfile.  I took the extra step of adding one for Ubuntu even-though we don't use it in CI (yet): 

https://github.com/containers/libpod/blob/master/Dockerfile.ubuntu

Let me know if that's not clear or doesn't make sense or you have questions.

**Comment by abitrolly** [comment]
@cevich in ideal world the list of OS level dependencies should be maintained in one central location, such as https://github.com/containers/libpod/blob/master/contrib/dependencies.txt and then reused for
* podman.io
* Makefile
* Dockerfile
* snap
* packer
* ansible and other provisioning tools if any are needed for tests

**Comment by cevich** [comment]
> in ideal world the list of OS level dependencies should be maintained 

Yep, that's the general idea here, except it isn't as simple as a package list.  We need a script because there are occasionally dependency-related actions we need to take depending on different circumstances in time, distro versions, etc.  For example, we found a major bug in BATS and needed to script a workaround for a specific version, since we couldn't wait for an upstream fix.

**Comment by cevich** [comment]
I can add a container image for Dockerfile.ubuntu and have it auto-build by `quay.io` if that would help you reference by a FROM.

**Comment by vrothberg** [comment]
Is this PR still desired?

**Comment by abitrolly** [comment]
I failed to run `snapcraft` on Fedora - https://forum.snapcraft.io/t/fedora-32-snapcraft-fail-multipass-timeout/17783/13 - so no progress from my side. `snap` support there is broken. There is no network for some reason. Neither for `snapcraft`, nor for LXD installed from snaps - containers can not ping google and I've tired fighting problems. I can not even run snapcraft with podman, because of SELinux bug with Python `shutil`.

**Pull request closed without merging**